code/modules/mob/living/living.dm

```diff
@@ -1167,3 +1167,36 @@
 	C.take_organ_damage(damage)
 	C.KnockDown(3 SECONDS)
 	C.visible_message("<span class='danger'>[C] crashes into [src], knocking them both over!</span>", "<span class='userdanger'>You violently crash into [src]!</span>")
+
+/mob/living/throw_impact(atom/hit_atom, throwingdatum)
+	. = ..()
+	if(has_status_effect(STATUS_EFFECT_CHARGING))
```

Comment:
You can probably early return here if it has either of the two status effects we check for.